src/test/java/org/mongojack/TestDBUpdateSerialization.java

```diff
@@ -182,4 +206,71 @@ public void serialize(String value, JsonGenerator jgen,
             }
         }
     }
+    
+    public static class NestedRepeatedAttributeName {
+    	public static class Inner {
+    		public Date timestamp;
+
+			@Override
+			public int hashCode() {
```

Comment:
Ugly autogenerated ones from eclipse. I'll just remove them and do attribute comparison after the fetch for validation. Easier to maintain anyway.